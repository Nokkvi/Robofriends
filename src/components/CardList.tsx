import React, {Component} from 'react';
import Card from './Card';

import { IRobot } from './interfaces.d';

class CardList extends Component<{robots: IRobot[]}, {}> {
  render() {
    const {robots} = this.props;
    return (
      <div>
        {
          robots.map((user: IRobot, i: number) => {
            return (
              <Card 
              key={i} 
              robot={user}
              />
            );
          })
        }
      </div>
    )
  }
}

export default CardList;