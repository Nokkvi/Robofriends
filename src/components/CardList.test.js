import { shallow } from 'enzyme';
import React from 'react';
import CardList from './CardList';

it('expects to render Card component', () => {
  const mockRobots = [
    {
      id: 1,
      name: 'John Smith',
      username: 'Johns',
      email: 'john@example.com'
    },
    {
      id: 2,
      name: 'John Snow',
      username: 'Knowsalot',
      email: 'john@knowsnothing.com'
    }
  ]
  expect(shallow(<CardList robots={mockRobots} />)).toMatchSnapshot();
})