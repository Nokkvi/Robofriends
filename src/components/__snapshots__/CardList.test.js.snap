// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expects to render Card component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardList
    robots={
        Array [
            Object {
              "email": "john@example.com",
              "id": 1,
              "name": "John Smith",
              "username": "Johns",
            },
            Object {
              "email": "john@knowsnothing.com",
              "id": 2,
              "name": "John Snow",
              "username": "Knowsalot",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          email="john@example.com"
          id={1}
          name="John Smith"
/>,
        <Card
          email="john@knowsnothing.com"
          id={2}
          name="John Snow"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "email": "john@example.com",
          "id": 1,
          "name": "John Smith",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "email": "john@knowsnothing.com",
          "id": 2,
          "name": "John Snow",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            email="john@example.com"
            id={1}
            name="John Smith"
/>,
          <Card
            email="john@knowsnothing.com"
            id={2}
            name="John Snow"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "email": "john@example.com",
            "id": 1,
            "name": "John Smith",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "email": "john@knowsnothing.com",
            "id": 2,
            "name": "John Snow",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
